lib/context-base.js

```diff
@@ -29,3 +39,7 @@ ContextBase.prototype.getScope = function() {
     method.ctor ||
     method.sharedMethod && method.sharedMethod.ctor;
 };
+
+ContextBase.prototype.setReturnArgByName = function(name, value) {
```

Comment:
Suggestion: I wonder whether if it will be useful if we move `function NOOP() {}` to the top of the file and call it wherever needed?